dash_labs/dashdown.py

```diff
@@ -0,0 +1,419 @@
+            "exec_code": exec_code,
+        }
+
+        _prohibited_props_check(props, prohibited_props)
+
+        # make a unique id for clipboard based on the markdown filename. Use UUID instead?
+        clipboard_props["id"] = (
+            filename.split(".")[0].replace("\\", "/").replace("/", "-")
+        )
+
+        # create layout
+        reconstructed = []
+        code_block = 0
+        for i, section in enumerate(split_notebook):
+            if "```" in section:
+                code_block += 1
+                app_card = ""
+                updated_props = _update_props(
+                    props, prohibited_props, section, code_block, filename
+                )
+
+                code_card = html.Div(
+                    [
+                        dcc.Markdown(section, **updated_props["code_markdown_props"]),
+                        dcc.Clipboard(
+                            target_id=f"{clipboard_props['id']}{i}",
+                            **updated_props["clipboard_props"],
+                        ),
+                    ],
+                    id=f"{clipboard_props['id']}{i}",
+                    style={"position": "relative"},
+                )
+
+                if updated_props["exec_code"]:
+                    if "app.callback" in section and "app" not in scope:
+                        raise Exception(
+                            """
+                            You must pass your own app object into the scope
+                            with scope={'app': app} if using callbacks.
+                            """
+                        )
+                    if not scope_creep:
+                        # make a copy
+                        code_scope = dict(**scope)
+                    else:
+                        # use the same dict that'll keep getting mutated
+                        code_scope = scope
+                    try:
+                        code = section.replace("```python", "").replace("```", "")
+                        app_card = _run_code(
+                            code,
+                            scope=code_scope,
+                            div_props=updated_props["app_div_props"],
+                        )
+                    except Exception as e:
+                        print(
+                            f"""
+                        The following error was generated while attempting to execute 
+                        code block number: {code_block} in file: {filename}
+                        error:  {e}                    
+                        """
+                        )
+                        pass
+
+                # side by side on large screens
+                lg = 6 if updated_props["side_by_side"] else 12
+                if updated_props["code_first"]:
+                    code_display = [
+                        dbc.Col(code_card, width=12, lg=lg),
+                        dbc.Col(app_card, width=12, lg=lg),
```

Comment:
let's remove the `dbc` dependency and use `html` and `style={}` 🙂

Reply:
Done